backend/simulators/aruba_simulator.py

```
@@ -0,0 +1,324 @@
+            socket_connect_timeout=2,
+        )
+        await client.ping()
+        return client
+    except Exception as exc:
+        logger.warning("Aruba sim: Redis unavailable (%s) — falling back to random clients", exc)
+        return None
+
+
+def _random_clients() -> List[Dict]:
+    """Return a randomised subset of device clients, each on a random AP."""
+    n = random.randint(10, min(40, len(SAMPLE_DEVICES)))
```

Ruff (0.15.9)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)